backend/apps/experiments/middleware.py

```
@@ -1,35 +1,69 @@
+                        experiment.id,
+                    )
+                    continue
+
+                if traffic_allocation is None:
+                    logger.warning(
+                        "Skipping experiment assignment: experiment %s has no traffic_allocation",
+                        experiment.id,
+                    )
+                    continue
+
+                if random.random() > traffic_allocation:
```

Ruff (0.15.21)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)